### ISSUE TITLE
setStatus method doesn't Exist.

### ISSUE THREAD

**Issue by LaserStony** [sequelize/sequelize#9275]
## What are you doing?
I Created a Project and a Status Model.  Combining them using certain associations. But, It says setStatus Doesn't Exist. Although I did Checked through Association Log, and It shows it does.. I think Something might be wrong with the Function and Stuff...

```js
//Sequelize Definition
const Sequelize = require("sequelize");
const sequelize = new Sequelize("database", "username", "password", {
	host: "localhost",
	dialect: "sqlite",
	// SQLite only
	//storage: "./data/database.db"
});
//CODE
this.Project = sequelize.define("Project", {
			id: {
				type: Sequelize.INTEGER,
				autoIncrement: true,
				primaryKey: true
			},
			name: {
				type: Sequelize.STRING,
				allowNull: false
			},
			abbr: {
				type: Sequelize.STRING,
				allowNull: false
			}
		});
this.Status = sequelize.define("Status", {
			id: {
				type: Sequelize.INTEGER,
				autoIncrement: true,
				primaryKey: true
			},
			name: {
				type: Sequelize.STRING,
				allowNull: false
			},
			description: {
				type: Sequelize.STRING,
				allowNull: true
			},
			abbr: {
				type: Sequelize.STRING,
				allowNull: false
			}
		});

    this.Project.hasMany(this.Chapter);
    this.Status.belongsTo(this.Project);
    this.Project.hasOne(this.Status);


//The Function I used to get Association Log
async listAssociates() {
		for (let assoc of Object.keys(this.Project.associations)) {
			for (let accessor of Object.keys(this.Project.associations[assoc].accessors)) {
				console.log(this.Project.name + "." + this.Project.associations[assoc].accessors[accessor] + "()");
			}
		}
	}
//WHERE the Error Occurs
await  client.db.Project.findOrCreate({
				where: {
					name: projectTitle,
					abbr: projectAbbr
				}
			}).then((prjct) => {
				
				client.db.Status.findOne({where: {abbr: Status}}).then(stat => {
					prjct.setStatus(stat); //This is the Line where the error points to.
				});
});
//ASSOCIATION LOG
Project.getChapters()
Project.setChapters()
Project.addChapters()
Project.addChapter()
Project.createChapter()
Project.removeChapter()
Project.removeChapters()
Project.hasChapter()
Project.hasChapters()
Project.countChapters()
Project.getStatus()
Project.setStatus()
Project.createStatus()

//ERROR LOG
Unhandled rejection TypeError: prjct.setStatus is not a function
    at client.db.Status.findOne.then.stat (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/helpers/dataholder.js:24:12)
    at tryCatcher (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/i10Studios/VisualCodeProjects/LSScanlationManagerBot/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

## What do you expect to happen?
I was Expecting it To Work

## What is actually happening?
It says it doesn't Exist Although, it shows in Association Log that it Does...

_Output, either JSON or SQL_

__Dialect: sqlite
__Dialect version:__ 4.0.0
__Database version:__ XXX
__Sequelize version:__ 4.37.6
__Tested with latest release:__ No (If yes, specify that version)

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

**Comment by sushantdhiman** [comment]
http://docs.sequelizejs.com/class/lib/model.js~Model.html#static-method-findOrCreate returns `[instance, created]`, use `spread` instead of `then` with `findOrCreate`

**Comment by LaserStony** [comment]
thanx @sushantdhiman. It was My Fault.. I checked and Now the Error Doesn't Exist But Still,  I cant get the Status that I set. I think there Might be something wrong with `Model.hasOne(AnotherModel);` function. as I am unable to get any thing. I am catching the Promise. and tried it with and Without is
```
projects[i].getStatus().then(stat =>{
								console.log(stat);
							});
```
it logs `null`, Although I set the setStatus(status) first and see in log that it is set.
